lib/compat/wordpress-6.2/site-editor.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Updates the site-editor.php file
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Remove home template setting for WP 6.2.
+ *
+ * @param array                   $settings Existing block editor settings.
+ * @param WP_Block_Editor_Context $context The current block editor context.
+ * @return array
+ */
+function gutenberg_site_editor_unset_homepage_setting( $settings, $context ) {
+	if ( isset( $context->post ) ) {
+		return $settings;
+	}
+
+	unset( $settings['__unstableHomeTemplate'] );
```

Comment:
Can you elaborate a bit about this change? If I remember correctly it was need for classic themes(I might be wrong though). Why we don't need this in this PR?

Reply:
I removed it because it didn't seem used to me at all.

Reply:
It was introduced in #38817 to solve a few issues at the time.

* Can you point me to changes when this logic became irrelevant?
* Is now the app's edited template state fully controlled by `useInitEditedEntityFromURL`?

Reply:
@youknowriad, let me know if any additional details are needed from me.

Reply:
> Is now the app's edited template state fully controlled by useInitEditedEntityFromURL?

Yes,

> Can you point me to changes when this logic became irrelevant?

I don't know where is this to be honest. I didn't feel like I needed this variable for everything to work properly. Maybe I'm missing something.

Reply:
I will test later today or Monday and try to provide better answers :)

Reply:
I tested, and the setting is no longer needed. Initially, we introduced it to have better error handling for WSOD described in #38817. But [server-side redirection](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/site-editor.php#L31-L46) should prevent similar WSOD.

Reply:
Thanks for checking.